src/main/java/org/petshelter/service/PetShelterServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package org.petshelter.service;
+
+
+import lombok.RequiredArgsConstructor;
+import org.petshelter.model.Pet;
+import org.petshelter.model.Shelter;
+
+import static java.lang.System.out;
+
+@RequiredArgsConstructor
+public class PetShelterServiceImpl implements PetShelterService {
+    private final Shelter shelter;
+
+    @Override
+    public void addPet(Pet newPet) {
+        try {
+            if (shelter.getPets() != null) {
+                String key = Integer.toString(shelter.getPets().size() + 1);
```

Comment:
this is incorrect. you don't know exactly the key value, so you can lose an access to an animal

Reply:
also duplicates of a key may occur which may cause a rewrites